packager/packager.cc

```diff
@@ -275,23 +275,121 @@ class PackageInfoImpl final : public core::packages::PackageInfo {
     explicit PackageInfoImpl(const PackageInfoImpl &) = default;
     PackageInfoImpl &operator=(const PackageInfoImpl &) = delete;
 
-    optional<core::AutocorrectSuggestion> addImport(const core::GlobalState &gs, const PackageInfo &pkg,
-                                                    bool isTestImport) const {
-        auto &info = PackageInfoImpl::from(pkg);
-        auto it = absl::c_find_if(importedPackageNames, [&info](auto &import) { return import.name == info.name; });
-        if (it != importedPackageNames.end()) {
-            if (!isTestImport && it->type == ImportType::Test) {
-                return convertTestImport(gs, info, core::Loc(fullLoc().file(), it->name.loc));
+    // What order should these packages be in the import list?
+    // Returns -1 if a should come before b, 0 if they are equivalent, and 1 if a should come after b.
+    // TODO(neil): sort by alphabetical order too
+    int orderByStrictness(const core::packages::PackageDB &packageDB, const PackageInfoImpl &a,
+                          const PackageInfoImpl &b) const {
+        if (!strictDependenciesLevel.has_value() || !a.strictDependenciesLevel.has_value() ||
```

Comment:
This docstring makes it sound like we're just comparing `a` and `b`...but if we're doing that, why are we checking `this->strictDependenciesLevel` in this function?  Shouldn't this function be static?

If that's not the case, how does the strictness of the current function matter for the relative sorting order of `a` and `b`?  (It seems like it shouldn't, but maybe there's some peculiar behavior of `gen-packages` that we're trying to replicate here?)

Reply:
Yup, it's a function of the strictness level. I added a comment explaining the ordering.